From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Sat, 19 Nov 2022 23:54:53 +0800
Subject: [PATCH] Maven publishing


diff --git a/build.gradle.kts b/build.gradle.kts
index 5c8dd4d3313a791d1fee00ec5d4bc595b76b7d6d..4d09fbd6e8205469cc438e922088e293a5dc6851 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -153,3 +153,24 @@ tasks.check {
     dependsOn(scanJar)
 }
 // Paper end
+
+// KeYi start
+publishing {
+    repositories {
+        maven {
+            name = "githubPackage"
+            url = uri("https://maven.pkg.github.com/KeYiMC/KeYi")
+
+            credentials.username = System.getenv("GITHUB_USERNAME")
+            credentials.password = System.getenv("GITHUB_TOKEN")
+
+        }
+
+        publications {
+            register<MavenPublication>("gpr") {
+                from(components["java"])
+            }
+        }
+    }
+}
+// KeYi end
\ No newline at end of file
