From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Sat, 19 Nov 2022 23:54:53 +0800
Subject: [PATCH] Maven publishing


diff --git a/build.gradle.kts b/build.gradle.kts
index 051b2db3548d1b24f591f05eece0c8241475e9bf..85e30cd20f0f8534724ce4e309e28ef0031f683c 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -145,3 +145,24 @@ tasks.check {
     dependsOn(scanJar)
 }
 // Paper end
+
+// KeYi start
+publishing {
+    repositories {
+        maven {
+            name = "githubPackage"
+            url = uri("https://maven.pkg.github.com/KeYiMC/KeYi")
+
+            credentials.username = System.getenv("GITHUB_USERNAME")
+            credentials.password = System.getenv("GITHUB_TOKEN")
+
+        }
+
+        publications {
+            register<MavenPublication>("gpr") {
+                from(components["java"])
+            }
+        }
+    }
+}
+// KeYi end
\ No newline at end of file
