From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Fri, 9 Dec 2022 23:13:43 +0800
Subject: [PATCH] Remove "Spark Profiler"

This reverts commit 2f3f630834d34ee625be79fac6bc429d0e10d929.

diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index 37b42f155f7e1f743ea49d957123f2dc05f3b3fd..ba869354adc59db2fc547c481c1ed4d5d0af23b7 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -167,12 +167,6 @@ public final class SimplePluginManager implements PluginManager {
                 server.getLogger().log(Level.SEVERE, "Could not load '" + file.getPath() + "' in folder '" + file.getParentFile().getPath() + "'", ex); // Paper
                 continue;
             }
-            // Purpur start
-            if (!Boolean.getBoolean("Purpur.IReallyDontWantSpark") && plugins.containsKey(description.getName()) && description.getName().equalsIgnoreCase("spark")) {
-                server.getLogger().log(Level.INFO, "Purpur: Using user-provided spark plugin instead of our own.");
-                continue;
-            }
-            // Purpur end
 
             File replacedFile = plugins.put(description.getName(), file);
             if (replacedFile != null) {
