From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Sat, 22 Oct 2022 17:52:46 +0800
Subject: [PATCH] Let users decide if we should warn while running in proxy
 mode


diff --git a/src/main/java/cc/keyimc/keyi/KeyiConfig.java b/src/main/java/cc/keyimc/keyi/KeyiConfig.java
index ed570a7d344507066ed106ba6ea3ceb806432d27..e4019f346c88acc6909589916f47dc8fe1dc5d18 100644
--- a/src/main/java/cc/keyimc/keyi/KeyiConfig.java
+++ b/src/main/java/cc/keyimc/keyi/KeyiConfig.java
@@ -135,9 +135,11 @@ public final class KeyiConfig {
 
     public static boolean enableRootUserWarning = true;
     public static boolean enableOfflineModeWarning = true;
+    public static boolean enableProxyUnsafeWarning = true;
 
     public static void misc() {
         enableRootUserWarning = getBoolean("misc.enable-root-user-warning", enableRootUserWarning);
         enableOfflineModeWarning = getBoolean("misc.enable-offline-mode-warning", enableOfflineModeWarning);
+        enableProxyUnsafeWarning = getBoolean("misc.enable-proxy-unsafe-warning", enableProxyUnsafeWarning);
     }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 1ff49a9074292207e01a7a07a42cff445cf96642..05c7380146cc3a8bb669432003034a3b0bd4cec4 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -333,7 +333,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
             }
             // KeYi end
             // Spigot start
-            if (io.papermc.paper.configuration.GlobalConfiguration.get().proxies.isProxyOnlineMode()) { // Purpur
+            if (io.papermc.paper.configuration.GlobalConfiguration.get().proxies.isProxyOnlineMode() && cc.keyimc.keyi.KeyiConfig.enableProxyUnsafeWarning) { // Purpur // KeYi - Let users decide if we should warn while running in proxy mode
                 DedicatedServer.LOGGER.warn("Whilst this makes it possible to use BungeeCord or Velocity, unless access to your server is properly restricted, it also opens up the ability for hackers to connect with any username they choose."); // Purpur
                 DedicatedServer.LOGGER.warn("Please see http://www.spigotmc.org/wiki/firewall-guide/ for further information.");
             } else {
